modules/core/src/main/java/org/locationtech/jts/geom/Geometry.java

```diff
@@ -1648,15 +1648,26 @@ public Object clone() {
   }
   
   /**
-   * Creates and returns a full copy of this {@link Geometry} object
-   * (including all coordinates contained by it).
-   * Subclasses are responsible for implementing this method and copying
-   * their internal data.
+   * Creates a deep copy of this {@link Geometry} object.
+   * Coordinate sequences contained in it are copied.
+   * All instance fields are copied (i.e. the <tt>SRID</tt> and <tt>userData</tt>).
    *
-   * @return a clone of this instance
+   * @return a deep copy of this geometry
    */
-  abstract public Geometry copy();
-
+  public Geometry copy() {
+    Geometry copy = copyInternal();
+    copy.SRID = this.SRID;
+    copy.userData = this.userData; 
```

Comment:
Is there an easy way to copy the userData?

Reply:
Not that I can see.  The userData is really under client control, so they need to copy it if needed. 

I will add a comment about this to the copy() Javadoc.

Reply:
+1, documenting it makes sense to me.